src/components/Editor/Footer.js

```diff
@@ -56,22 +55,34 @@ class SourceFooter extends PureComponent<Props, State> {
   constructor() {
     super();
 
-    this.state = { cursorPosition: { line: 1, column: 1 } };
+    this.state = { cursorPosition: { line: 0, column: 0 } };
   }
 
-  componentDidMount() {
-    const { editor } = this.props;
-    editor.codeMirror.on("cursorActivity", this.onCursorChange);
+  componentDidUpdate() {
+    const eventDoc = document.querySelector(".CodeMirror");
+    const codeMirror = this.getCodeMirror(eventDoc);
+    if (eventDoc) {
+      codeMirror.on("cursorActivity", this.onCursorChange);
+    }
   }
 
   componentWillUnmount() {
-    const { editor } = this.props;
-    editor.codeMirror.off("cursorActivity", this.onCursorChange);
+    const eventDoc = document.querySelector(".CodeMirror");
+    const codeMirror = this.getCodeMirror(eventDoc);
+    codeMirror.off("cursorActivity", this.onCursorChange);
+  }
+
+  getCodeMirror(eventDoc: Object): Object {
```

Comment:
Having this method just to return a property on its argument doesn't seem very useful.

Reply:
Also if the querySelector returned null and you pass that value to getCodeMirror, you will get an error.